holidays/countries/saint_kitts_and_nevis.py

```diff
@@ -0,0 +1,188 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/python-holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.calendars.gregorian import FEB, MAR, APR, JUL, AUG, SEP, DEC, SUN
+from holidays.constants import HALF_DAY, PUBLIC, WORKDAY
+from holidays.groups import ChristianHolidays, InternationalHolidays, StaticHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_MON, SUN_TO_NEXT_TUE
+
+
+class SaintKittsAndNevis(
+    ObservedHolidayBase, ChristianHolidays, InternationalHolidays, StaticHolidays
+):
+    """
+    References:
+      - https://lawcommission.gov.kn/wp-content/documents/Revised-Acts-of-St-Kitts-and-Nevis/Revised-Acts-of-St-Kitts-and-Nevis-2009/Ch-23_23-Public-Holidays-Act.pdf
+      - https://web.archive.org/web/20220124000224/https://aglcskn.info/wp-content/documents/Act02and09TOC/Ch-23_23-Public-Holidays-Act.pdf
+      - https://en.wikipedia.org/wiki/Public_holidays_in_Saint_Kitts_and_Nevis
+
+    Cross-Checked With:
+      - https://sknhcottawa.gov.kn/in-skn-national-public-holidays/
+      - https://www.timeanddate.com/holidays/saint-kitts-and-nevis/
+
+    If Sovereign’s Birthday, New Year's Day, Independence Day, or National Heroes Day
+    fall on a Sunday the next following Monday shall be a public holiday.
+
+    Boxing Day—that is the day after Christmas Day, but if Christmas Day falls
+    on a Saturday, then the next following Monday shall be a public holiday, and if
+    Christmas Day falls on a Sunday, then the next following Monday and Tuesday
+    shall be public holidays.
+
+    While Culturama Day (first started in 1974) and Carnival Day are never officially
+    included in the main Chapter 23.23 document, they're de facto added since at least
+    2015 and should be considered as such.
+    """
+
+    country = "KN"
+    supported_categories = (HALF_DAY, PUBLIC, WORKDAY)
+    # %s (observed).
+    observed_label = "%s (observed)"
+    weekend = {SUN}
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, SaintKittsAndNevisStaticHolidays)
+        kwargs.setdefault("observed_rule", SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Public Holidays Act, Act 7 of 1983 Amendment.
+        if self._year <= 1982:
+            return None
+
+        # Carnival Day.
```

Comment:
Are you sure about Carnival Day on January, 1?

Reply:
Yes, I'm following the official naming convention as used by Saint Kitts and Nevis government here - see the test case references for the year 2021-2024 in the [test_saint_kitts_and_nevis.py](https://github.com/vacanza/python-holidays/blob/557132683d975aea178c3bdc1483e8154c089272/tests/countries/test_saint_kitts_and_nevis.py).

Reply:
Yes, indeed. But it's a bit unusual...